Cleaner branch

micha7a opened the pull request:
Clean most of theory related stuff. 

Stuff left to do:
 - remove unused folders 
 - add figures to GenerateAllFigures

duembgen commented:
I added the figures used to create Figure 3. The three eps figures are obtained by saving each layer individually as an eps. Do we also want the final pdf of the figure here?

micha7a commented:
> Do we also want the final pdf of the figure here?
I added .png files because it's tricky to display .pdf or .eps in the notebook/markdown/html. 

I also removed some of the results, but I think there is more to remove.

micha7a commented:
Aaagh, more stuff, sorry. So, I forgot to add one script (`generate_noisy_results.py`) and I also noticed that `generate_results.py` is a module, not a script. Should we move it?

duembgen commented on GUIDELINES.md:
```diff
@@ -76,4 +77,19 @@ add those as github submodules, for example:
+ - Use `NotImplementError` either for functions that are under development (as in "not yet implemented") or for
+  abstract classes. For an option/flag that is not supported use `ValueError`.
+
+
```
Nice!
